pkg/stanza/fileconsumer/file_test.go

```diff
@@ -362,14 +366,16 @@ func TestSymlinkedFiles(t *testing.T) {
 	sink.ExpectNoCalls(t)
 
 	// Create and update symlink to each of the files over time.
-	for i := 1; i <= numFiles; i++ {
-		targetLogFilePath := filepath.Join(tempDir, fmt.Sprintf("%d.log", i))
+	for targetLogFilePath, expectedTokens := range expectedTokenMap {
 		require.NoError(t, os.Symlink(targetLogFilePath, symlinkFilePath))
-		// The sleep time here must be larger than the poll_interval value
-		time.Sleep(pollInterval + 1*time.Millisecond)
+
+		require.Eventually(t, func() bool {
+			sink.ExpectTokens(t, expectedTokens...)
+			return true
+		}, 3*testTickInterval, testTickInterval)
```

Comment:
The rapidness of this appears to be part of what the test was originally intended for, but due to the flake rate I felt it needed slightly more wiggle room. I settled on 3 ticks to expect the tokens to be read. We'll have to keep an eye out to see if this is too lenient.